src/SystemView.cpp

```diff
@@ -28,26 +30,106 @@ static const float WHEEL_SENSITIVITY = .1f;		// Should be a variable in user set
 // i don't know how to name it
 static const double ROUGH_SIZE_OF_TURD = 10.0;
 
-TransferPlanner::TransferPlanner() {
+TransferPlanner::TransferPlanner() :
+	m_position(0., 0., 0.), m_velocity(0., 0., 0.)
+{
 	m_dvPrograde = 0.0;
 	m_dvNormal = 0.0;
 	m_dvRadial = 0.0;
+	m_startTime = 0.0;
 	m_factor = 1;
 }
 
-vector3d TransferPlanner::GetVel() { return Pi::player->GetVelocity() + GetOffsetVel(); }
+vector3d TransferPlanner::GetVel() const { return m_velocity + GetOffsetVel(); }
+
+vector3d TransferPlanner::GetOffsetVel() const {
+	if(m_position.ExactlyEqual(vector3d(0., 0., 0.)))
+		return vector3d(0., 0., 0.);
+
+	const vector3d pNormal = m_position.Cross(m_velocity);
+
+	return m_dvPrograde * m_velocity.Normalized() +
+		   m_dvNormal   * pNormal.Normalized() +
+		   m_dvRadial   * m_position.Normalized();
+}
+
```

Comment:
I got some troubles because I did not get exactly how to format the code ;)
I found different approaches for curly braces, and... I just choose what I thought it was the "usual" way (even if I generally use braces on the same column as well).
What do you think about fixing some rules to use with astyle and making a huge commit with code format refactory?